site/content/en/docs/user/stages-configuration.md

```diff
@@ -138,6 +139,45 @@ if a "Change Stage" is matched. The update action itself consequently causes a n
 However, this event-driven approach to applying Stages has a limitation: `kwok` won’t apply a Stage until a new event associated with that resource is received. To address the limitation,
 users can utilize the `immediateNextStage` field to make the controller apply Stages immediately rather than waiting for an event pushed from the apiserver.
 
+## How Delay is Calculated
+
+The delay time of applying a Stage is obtained by adding a constant time period and a randomized interval,
+which will be denoted by *duration* and *jitter* respectively in the following.
+
+- `durationMilliseconds`: provides *duration*.
+- `jitterDurationMilliseconds`: calculates *jitter* by *random*(`jitterDurationMilliseconds`-*duration*).
+  It shall be larger than `durationMilliseconds` if you want to inject jitter to the delay.
+  Otherwise, it will be used directly as the delay time without any randomization.
+
+`kwok` also provides other fields to flexibly handle more situations. They are used to extract
+and parse a RFC3339 timestamp field of a resource to help determine the delay time dynamically.
+
+- `durationFrom`: calculates *duration* by `durationFrom`-*now*
+- `jitterDurationFrom`: calculates *jitter* by *random*(`jitterDurationFrom`-*now*-*duration*),
+  where *now* is the timestamp when the delay starts.
+
+{{< hint "info" >}}
+`durationFrom` and `jitterDurationFrom` have a higher priority than `durationMilliseconds`
+and `jitterDurationMilliseconds` if both are set.
+{{< /hint >}}
+
+Let’s explain a little bit about the motivation behind these two advanced fields.
+But before that, for a better understanding, we briefly describe how kubelet "delete" a pod from a node.
+
+Here are the steps to remove a pod in Kubernetes:
+
+1. Execute the command kubectl delete pod. The apiserver receives the deletion request but does not immediately remove the corresponding pod resource from etcd.
```

Comment:
```suggestion
1. Execute the command `kubectl delete pod`. The apiserver receives the deletion request but does not immediately remove the corresponding pod resource from etcd.
```

Reply:
I was too careless... Thanks again for your patience! @windsonsea @wzshiming

Reply:
All right, admire your contributions 👍